rtos/tests/UNITTESTS/doubles/CMakeLists.txt

```diff
@@ -9,17 +9,15 @@ target_include_directories(mbed-headers-rtos
         ${mbed-os_SOURCE_DIR}/rtos/include/rtos
 )
 
-add_library(mbed-stubs-rtos-headers INTERFACE)
+add_library(mbed-stubs-rtos)
+
+add_definitions(-DUNITTEST)
```

Comment:
Thanks for your review comment. This add_definition is required, and we can remove the one in "UNITTEST/CMakeLists.txt" as add_definitions (definition of UNITTEST macro ) visible in the current directory CMake and below add_subdirectories from there and don't have the option to expose to parent scope to visible across CMake

Reply:
We may want to create a "unit-tests" INTERFACE target in UNITTEST/CMakeLists.txt, and link to it in each test executable, instead of using the directory scoped `add_definitions` to set this macro. Or maybe we can just bodge it into `CMAKE_C_FLAGS` and `CMAKE_CXX_FLAGS` globally, like we do with the coverage compile options?

Reply:
> Or maybe we can just bodge it into `CMAKE_C_FLAGS` and `CMAKE_CXX_FLAGS` globally, like we do with the coverage compile options?

This sounds like the easiest approach.

A more fundamental question is, a few libraries check `UNITTEST`:
```
$ grep -rn '\bUNITTEST\b'
drivers/include/drivers/MbedCRC.h:32:#ifdef UNITTEST
UNITTESTS/target_h/platform/mbed_assert.h:44:#ifndef UNITTEST
storage/kvstore/tdbstore/tests/UNITTESTS/TDBStore/CMakeLists.txt:10:        UNITTEST
storage/kvstore/filesystemstore/tests/UNITTESTS/FileSystemStore/CMakeLists.txt:10:        UNITTEST
rtos/include/rtos/internal/mbed_rtos_storage.h:20:#if MBED_CONF_RTOS_PRESENT || defined(UNITTEST)
rtos/include/rtos/internal/mbed_rtos1_types.h:20:#if MBED_CONF_RTOS_PRESENT || defined(UNITTEST)
rtos/include/rtos/Thread.h:36:#if MBED_CONF_RTOS_PRESENT || defined(DOXYGEN_ONLY) || defined(UNITTEST)
rtos/include/rtos/mbed_rtos_types.h:20:#if MBED_CONF_RTOS_PRESENT || defined(DOXYGEN_ONLY) || defined(UNITTEST)
```
but is this actually ideal? I thought libraries shouldn't need to specially handle tests. Instead, fakes/mocks/stubs/... should wrap tested modules while maintaining normal-use interfaces?
Anyway this is just out of curiosity, not the scope of this PR.

Reply:
Not ideal at all IMO. Adding test specific logic to your production/library code is the wrong approach. Ideally we'd remove the need for those checks altogether.

Reply:
There's one simple way to get rid of `-DUNITTEST`, at least from rtos which this PR is for. Notice the snippet in my previous comment - `MBED_CONF_RTOS_PRESENT` and `defined(UNITTEST)` are connected by `||`. So what about defining `MBED_CONF_RTOS_PRESENT=1` directly?

It's enough to remove `add_definitions(-DUNITTEST)` and add
```cmake
target_compile_definitions(mbed-stubs-rtos
    PRIVATE
        MBED_CONF_RTOS_PRESENT=1
)
```
here, and also add `MBED_CONF_RTOS_PRESENT=1` to https://github.com/ARMmbed/mbed-os/blob/8d3fc339652ae7294f76c75f029767d2db21a358/connectivity/cellular/tests/UNITTESTS/framework/device/athandler/CMakeLists.txt#L8-L16
because `ATHandler.cpp` specifically checks this macro.

Then we can safely remove all `defined(UNITTEST)` checks from `rtos/`.

Reply:
Seems like we should review all of these and fix them in another PR. It would be good to keep this PR scoped to relocation of stub libraries.

Reply:
There are not many of files to fix and the fix is simple as shown above, we can raise another PR to fix all of them in one go. I already made locally changes when experimenting a bit for my previous comment, so I can do it.

Reply:
> 
> 
> We may want to create a "unit-tests" INTERFACE target in UNITTEST/CMakeLists.txt, and link to it in each test executable, instead of using the directory scoped `add_definitions` to set this macro. Or maybe we can just bodge it into `CMAKE_C_FLAGS` and `CMAKE_CXX_FLAGS` globally, like we do with the coverage compile options?

I just moved the add_defenition into Mbed OS root Cmake under testing check as we may end up removing mbed-os/UNITTEST CMake after we migrated all stubs to their respective component and append -DUNITTEST into  CMAKE_C_FLAGS and CMAKE_CXX_FLAGS  will not work as it accepts -Dmacro=value when set() calls

Reply:
Either is fine for this PR. I've created an issue #14883 to fully remove the `UNITTEST` macro.